flight_controller: add KakuteH7v2 and KakuteH7mini (#2247)
PX4/PX4-user_guide: en/flight_controller/kakuteh7mini.md

# Holybro Kakute H7 mini

:::warning
PX4 does not manufacture this (or any) autopilot.
Contact the [manufacturer](https://shop.holybro.com/) for hardware support or compliance issues.
:::

> The [Holybro Kakute H7 mini](http://www.holybro.com/product/kakute-h7-mini/) flight controller is full of features including integrated Bluetooth, HD camera plug, dual plug-and-play 4in1 ESC ports, 9V VTX ON/OFF Pit Switch, barometer, OSD, 6x UARTs, 128MB Flash for Logging (not supported with PX4 yet), 5V and 9V BEC, and bigger soldering pad with easy layout and much more.

Reviewer: The feature list here is the same for both of the new vehicles you added. Is this correct?

Author: I don't really know. @vincentpoont2 would know better.

Reviewer 2: Difference are both not supported in PX4: 
1) Bluetooth
2) VTX Pit Switch

> The [Holybro Kakute H7 mini](http://www.holybro.com/product/kakute-h7-mini/) flight controller is full of features including integrated Bluetooth, HD camera plug, dual plug-and-play 4in1 ESC ports, 9V VTX ON/OFF Pit Switch, barometer, OSD, 6x UARTs, 128MB Flash for Logging (not supported with PX4 yet), 5V and 9V BEC, and bigger soldering pad with easy layout and much more.

Reviewer: " (not supported with PX4 yet)" - is logging still not supported?

```suggestion
The [Holybro Kakute H7 mini](http://www.holybro.com/product/kakute-h7-mini/) flight controller is full of features including integrated Bluetooth, HD camera plug, dual plug-and-play 4in1 ESC ports, 9V VTX ON/OFF Pit Switch, barometer, OSD, 6x UARTs, 128MB Flash for logging (not supported with PX4 yet), 5V and 9V BEC, and bigger soldering pad with easy layout and much more.
```

Author: It's not planned to be supported for now.

> The [Holybro Kakute H7 mini](http://www.holybro.com/product/kakute-h7-mini/) flight controller is full of features including integrated Bluetooth, HD camera plug, dual plug-and-play 4in1 ESC ports, 9V VTX ON/OFF Pit Switch, barometer, OSD, 6x UARTs, 128MB Flash for Logging (not supported with PX4 yet), 5V and 9V BEC, and bigger soldering pad with easy layout and much more.

Reviewer: Yes it has lots of features, but is it intended for a particular use case? I.e. racers etc? i.e. what would you use mini vs h7 for?

This would be better worded as something like: "This is a flight controller intended for racers. It has many features, including:"

Author: That would be good. @vincentpoont2

Reviewer: @vincentpoont2 @julianoes Progress ?

Reviewer 2: @hamishwillee I will review this today.

Reviewer 2: It is intended for Racer, but I don't think a lot of racer use PX4. I think it is good for people wanting to stay light weight and keep the the frame small.

The Kakute H7 mini builds upon the best features of its predecessor, the [Kakute F7](../flight_controller/kakutef7.md), and the [Kakute H7](../flight_controller/kakuteh7.md).

The board also has an on-board barometer, LED & buzzer pad, and I2C pad (SDA & SCL) for external GPS/magnetometers.

> ![Kakute H7mini top](../../assets/flight_controller/kakuteh7mini/kakuteh7mini_top.jpg)

Reviewer: FYI I shrunk the images.

Author: Ok, thanks.

| B+ | Battery positive voltage (2S-8S) ||
| SDA, SCL | I2C connection (for peripherals) ||
| 5V | 5V output (2A max) ||
| 3V3 | 3.3V output (0.25A max) ||
| VI | Video input from FPV camera ||
| VO | Video output to video transmitter ||
| CAM | To camera OSD control ||
| G or GND | Ground ||
| RSI | Analog RSSI (0-3.3V) input from receiver ||
| R1, T1 | UART1 RX and TX | TELEM1 |
| R3, T3 | UART3 RX and TX | NuttX debug console |
| R4, T4 | UART4 RX and TX | GPS1 |
| R6, T6 | UART6 RX and TX (R6 also located in the GH plug)| RC port |
| R7 | UART7 RX (RX is located in the plugs for use with 4-in-1 ESCs) |DShot telemetry|
| LED | WS2182 addressable LED signal wire (not tested) ||
| Z- | Piezo buzzer negative leg (Connect buzzer positive leg to 5V pad) ||
| M1 to M4 | Motor signal outputs (located in plug for use in 4-in-1 ESCs) ||
| M5 to M8 | Motor signal outputs (located in plug for use in 4-in-1 ESCs) ||
| Boot | Bootloader button | |

<a id="bootloader"></a>
## PX4 Bootloader Update

TODO: hex file

The board comes pre-installed with [Betaflight](https://github.com/betaflight/betaflight/wiki).
Before the PX4 firmware can be installed, the *PX4 bootloader* must be flashed.

> Download the [holybro_kakuteh7mini_bootloader.hex](TODO) bootloader binary and read [this page](../advanced_config/bootloader_update_from_betaflight.md) for flashing instructions.

Reviewer: Note, that for this to work you will have to reference the final version of the raw file on github rather than just link. Ditto for other places.

## Building Firmware

To [build PX4](../dev_setup/building_px4.md) for this target:

```
make holybro_kakuteh7mini_default
```

## Installing PX4 Firmware

The firmware can be installed in any of the normal ways:

- Build and upload the source:

  ```
  make holybro_kakuteh7mini_default upload
  ```
- [Load the firmware](../config/firmware.md) using *QGroundControl*.
  You can use either pre-built firmware or your own custom firmware.

:::note
KakuteH7mini is supported with PX4 main and v1.14 or newer.
:::

## PX4 Configuration

In addition to the [basic configuration](../config/README.md), the following parameters are important:

Parameter | Setting
--- | ---
[SYS_HAS_MAG](../advanced_config/parameter_reference.md#SYS_HAS_MAG) | This should be disabled since the board does not have an internal mag. You can enable it if you attach an external mag.

> [MOT_ORDERING](../advanced_config/parameter_reference.md#MOT_ORDERING) | If you use a 4-in-1 ESC with Betaflight motor assignment, this parameter can be set accordingly. 

Reviewer: This param no longer exists and needs to be removed/updated everywhere. 

I "think" the instructions are simply to use normal actuator configuration for the betaflight ESC, but there might be some trick to it - I've looked for notes on this but I can't find them. Perhaps ask @bkueng

Reviewer 3: Yes with the control allocation this parameter got obsolete.

Reviewer: Thanks. Actuator tuning is part of basic configuration, so I am just deleting this parameter.
```suggestion
```

# Holybro Kakute H7 mini

:::warning
PX4 does not manufacture this (or any) autopilot.
Contact the [manufacturer](https://shop.holybro.com/) for hardware support or compliance issues.
:::

> The [Holybro Kakute H7 mini](http://www.holybro.com/product/kakute-h7-mini/) a flight controller intended for lightweight build (Such as racers, etc.)

Reviewer: ```suggestion
The [Holybro Kakute H7 mini](http://www.holybro.com/product/kakute-h7-mini/) flight controller is intended for lightweight frame builds (such as racers, etc.).
```

>  This flight controller is full of features including integrated Bluetooth, HD camera plug, dual plug-and-play 4in1 ESC ports, 9V VTX ON/OFF Pit Switch, barometer, OSD, 6x UARTs, 128MB Flash for logging (not supported with PX4 yet), 5V BEC, and bigger soldering pad with easy layout and much more.

Reviewer: @vincentpoont2 In your comment in https://github.com/PX4/PX4-user_guide/pull/2247#discussion_r1099020115 , I think you are saying that bluetooth and the Pit switch are no on the mini - right?

```suggestion
 This flight controller has many useful features, including integrated HD camera plug, dual plug-and-play 4in1 ESC ports, barometer, OSD, 6x UARTs, 128MB Flash for logging (not supported with PX4 yet), 5V BEC, and bigger soldering pad with easy layout and much more.
```

Reviewer: @vincentpoont2 This is waiting on final check for you here!

Reviewer 2: @hamishwillee Sorry, it has the VTX pit switch, but no bluetooth. I have made the edit.

The Kakute H7 mini builds upon the best features of its predecessor, the [Kakute F7](../flight_controller/kakutef7.md), and the [Kakute H7](../flight_controller/kakuteh7.md).

The board also has an on-board barometer, LED & buzzer pad, and I2C pad (SDA & SCL) for external GPS/magnetometers.

<img src="../../assets/flight_controller/kakuteh7mini/kakuteh7mini_top.jpg" width="300px" title="KakuteH7Mini Top Image" /> <img src="../../assets/flight_controller/kakuteh7mini/kakuteh7mini_bottom.jpg" width="300px" title="KakuteH7Mini Bottom Image" />

:::note
This flight controller is [manufacturer supported](../flight_controller/autopilot_manufacturer_supported.md).
:::

:::note
PX4 runs on the H7 mini v1.3 and later.
:::

## Key Features

- MCU: STM32H743 32-bit processor running at 480 MHz
- IMU: BMI270
- Barometer: BMP280

> - OSD: AT7456E
> - VTX On/Off Pit Switch: Not used with PX4
> - 6x UARTs (1,2,3,4,6,7; UART2 is used for Bluetooth telemetry)

Reviewer: @vincentpoont2 In your comment in https://github.com/PX4/PX4-user_guide/pull/2247#discussion_r1099020115 , I think you are saying that bluetooth and the Pit switch are not on the mini - right? So this change is needed.

```suggestion
- OSD: AT7456E
- 6x UARTs (1,2,3,4,6,7)
```

Reviewer 2: Sorry, it has the VTX pit switch, but no bluetooth. I have made the edit.

# Holybro Kakute H7 mini

:::warning
PX4 does not manufacture this (or any) autopilot.
Contact the [manufacturer](https://shop.holybro.com/) for hardware support or compliance issues.
:::

The [Holybro Kakute H7 mini](http://www.holybro.com/product/kakute-h7-mini/) a flight controller intended for lightweight build (Such as racers, etc.)

 This flight controller is full of features including integrated Bluetooth, HD camera plug, dual plug-and-play 4in1 ESC ports, 9V VTX ON/OFF Pit Switch, barometer, OSD, 6x UARTs, 128MB Flash for logging (not supported with PX4 yet), 5V BEC, and bigger soldering pad with easy layout and much more.

The Kakute H7 mini builds upon the best features of its predecessor, the [Kakute F7](../flight_controller/kakutef7.md), and the [Kakute H7](../flight_controller/kakuteh7.md).

The board also has an on-board barometer, LED & buzzer pad, and I2C pad (SDA & SCL) for external GPS/magnetometers.

> <img src="../../assets/flight_controller/kakuteh7mini/kakuteh7mini_top.jpg" width="300px" title="KakuteH7Mini Top Image" />
> <img src="../../assets/flight_controller/kakuteh7mini/kakuteh7mini_bottom.jpg" width="300px" title="KakuteH7Mini Bottom Image" />

Reviewer: ```suggestion
<img src="../../assets/flight_controller/kakuteh7mini/kakuteh7mini_top.jpg" width="300px" title="KakuteH7Mini Top Image" /> <img src="../../assets/flight_controller/kakuteh7mini/kakuteh7mini_bottom.jpg" width="300px" title="KakuteH7Mini Bottom Image" />
```

## Connectors and Pins

| Pin              | Function                              | PX4 default |
| ---------------- | ------------------------------------- | ----------- |
| B+ | Battery positive voltage (2S-6S) ||
| VTX+ | Battery positive voltage (2S-6S) ||
| SDA, SCL | I2C connection (for peripherals) ||
| 5V | 5V output (2A max) ||
| 3V3 | 3.3V output (0.25A max) ||
| VI | Video input from FPV camera ||
| VO | Video output to video transmitter ||
| CAM | To camera OSD control ||
| G or GND | Ground ||
| RSI | Analog RSSI (0-3.3V) input from receiver ||
| R1, T1 | UART1 RX and TX | TELEM1 |
| R3, T3 | UART3 RX and TX | NuttX debug console |
| R4, T4 | UART4 RX and TX | GPS1 |
| R6, T6 | UART6 RX and TX (R6 also located in the GH plug)| RC port |
| R7 | UART7 RX (RX is located in the plugs for use with 4-in-1 ESCs) |DShot telemetry|
| LED | WS2182 addressable LED signal wire (not tested) ||
| Z- | Piezo buzzer negative leg (Connect buzzer positive leg to 5V pad) ||
| M1 to M4 | Motor signal outputs (located in plug for use in 4-in-1 ESCs) ||
| M5 to M8 | Motor signal outputs (located in plug for use in 4-in-1 ESCs) ||
| Boot | Bootloader button | |

<a id="bootloader"></a>

> ## PX4 Bootloader Update
> 
> TODO: hex file
> 
> The board comes pre-installed with [Betaflight](https://github.com/betaflight/betaflight/wiki).

Reviewer: ```suggestion
## PX4 Bootloader Update

The board comes pre-installed with [Betaflight](https://github.com/betaflight/betaflight/wiki).
```

Before the PX4 firmware can be installed, the *PX4 bootloader* must be flashed.
Download the [holybro_kakuteh7mini_bootloader.hex](https://github.com/PX4/PX4-user_guide/raw/main/assets/flight_controller/kakuteh7mini/holybro_kakuteh7mini_bootloader.hex) bootloader binary and read [this page](../advanced_config/bootloader_update_from_betaflight.md) for flashing instructions.

## Building Firmware

To [build PX4](../dev_setup/building_px4.md) for this target:

```
make holybro_kakuteh7mini_default
```

## Installing PX4 Firmware

:::note

> KakuteH7 is supported with PX4 master & PX4 v1.13 or newer. If you are loading the pre-built firmware via QGroundcontrol, you must use QGC Daily or QGC version newer than 4.1.7.
> 
> From PX4 v1.14 (or newer) *QGroundControl* will automatically find the default release, beta and development firmware. 
> Prior to that release you will need to manually build and install the firmware.

Reviewer: ```suggestion
KakuteH7 is supported with PX4 master & PX4 v1.13 or newer. If you are loading the pre-built firmware via QGroundcontrol, you must use QGC Daily or QGC version newer than 4.1.7.
Prior to that release you will need to manually build and install the firmware.
```
